### DISCUSSION TITLE
Reading a file in next.js vercel deployment

### DISCUSSION THREAD

**Discussion by FahadSalamatRandhawa** [vercel/community#3763] [Help]
### Summary

I have the following code and I am trying to read a file, basically file handling using 'fs'. It works fine locally but gives error when code is deployed to vercel. I have tried various methods, staticpath, process.cwd(), __dirName but doesn't seem to be working. Following is code

`if(check_user.rows.length<1){
            return new NextResponse(JSON.stringify({message:'incorrect info'}),{status:400})
        }
        const certsPath=path.join(process.cwd(),'/certs')
        const private_key=fs.readFileSync(certsPath+'/private.pem','utf8')
        const public_key=fs.readFileSync(certsPath+'public.pem','utf8')
        const public_hash=crypto.createHash('sha256').update(public_key).digest('base64')
        console.log(public_hash)
        console.log('Before JWT')
        try{
            const token=jwt.sign({ data:email }, private_key, { algorithm:'RS256' , keyid:public_hash })
            console.log(token)
            myCookies.set('jwt',token)
            console.log('After JWT')
        }catch(e){
            console.log(e)
            return new NextResponse('Error',{status:400})
        }
`
![JWTError](https://github.com/vercel/community/assets/66182071/333a344e-6197-494f-a05c-18e380699172)

### Example

https://github.com/FahadSalamatRandhawa?tab=repositories

### Steps to Reproduce

Try to login, the signin API reads private.pem file and created a JWT. Works fine on localhost:3000 but error on vercel
[https://jwt-with-zero-dev-jwt.vercel.app/](url)

**Comment by amyegan** [comment]
Hey @FahadSalamatRandhawa. I'm assuming that you use the CLI to deploy from local because the file really would be missing if the project was being deployed via GitHub. In that case, you need to change the the path for `readFileSync` from `/certs` to `/src/certs` since that is the location relative to the root of the project. 

However, you should consider moving the cert to an environment variable. You can [create a new environment variable in the UI](https://vercel.com/docs/projects/environment-variables#declare-an-environment-variable) and paste in the contents of `private.pem` as the value. That would give you a bit more deployment flexibility and remove the risk of the key being accidentally pushed to the public repo.

**Reply by FahadSalamatRandhawa** [reply]
https://github.com/FahadSalamatRandhawa/jwt_with_zeroDevJWT/tree/main

The certs are inside src, and I am deploying through github push. Although I've noticed the pem files are gone and env file is still there
